src/wazuh.py

```diff
@@ -357,54 +365,83 @@ def _generate_cluster_snippet(
 
 
 def change_api_password(username: str, old_password: str, new_password: str) -> None:
-    """Change Wazuh's API password for the default 'wazuh' user.
+    """Change Wazuh's API password for a given user.
 
     Args:
         username: the username to change the user for.
         old_password: the old API password for the user.
         new_password: the new API password for the user.
 
     Raises:
+        WazuhAuthenticationError: if an authentication error occurs.
         WazuhInstallationError: if an error occurs while processing the requests.
     """
     # The certificates might be self signed and there's no security hardening in
     # passing them to the request since tampering with `localhost` would mean the
     # container filesystem is compromised
     try:
-        r = requests.get(  # nosec
+        response = requests.get(  # nosec
             "https://localhost:55000/security/user/authenticate",
             auth=(username, old_password),
             timeout=10,
             verify=False,
         )
         # The old password has already been changed. Nothing to do.
-        if r.status_code == 401:
-            return
-        token = r.json()["data"]["token"]
-        headers = {
-            "Authorization": f"Bearer {token}",
-            "Content-Type": "application/json",
-        }
-        r = requests.get(  # nosec
+        if response.status_code == 401:
+            raise WazuhAuthenticationError("The provided password is not valid.")
+        response.raise_for_status()
+        token = response.json()["data"]["token"] if response.json()["data"] else None
```

Comment:
Might be interesting to log a debug message if the token is empty as the consequence will be that next call with raise a 401 which might either mean there's no token or the token didn't work